dev-client/src/context/ConnectivityContext.tsx

```diff
@@ -0,0 +1,96 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+import {createContext, useEffect, useRef, useState} from 'react';
+import {AppState} from 'react-native';
+
+import {
+  addEventListener,
+  NetInfoSubscription,
+} from '@react-native-community/netinfo';
+
+export type ConnectivityContextType = {
+  isOffline: boolean | null;
+  setIsOffline: (newValue: boolean | null) => void;
```

Comment:
IMO we shouldn't be exposing the state setter via the context like this; if possible it'd make more sense to be read-only, I think. (Unless there's a specific situation you've envisioned for setting it outside of this context provider?)

Reply:
Ah, good call, thanks!